### DISCUSSION TITLE
ERROR: Repository has unknown error. "message" does not match policy-specified pattern

### DISCUSSION THREAD

**Discussion by snowleopard24** [renovatebot/renovate#22307] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

Azure DevOps

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm trying to configure Renovate for my repository and am likely missing some configuration and the error message suggests that it is trying to push to https://**redacted**@whitesourcesoftware.com'/ and then fails to push some refs to my repo. So, wanted to check what configuration is required to get this to work in Azure DevOps pipeline

Environment:
1. Azure DevOps
2. RenovateMe extension v 35.95.1
3. Using extension's built-in task in pipeline yml with renovate.json at the root
   - task: RenovateMe@1
         inputs:
         renovateOptionsVersion: 'latest'

### What you tried so far.

```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [":semanticCommits", ":dependencyDashboard", ":dependencyDashboardApproval"],
    "azure-pipelines": {
      "enabled": true
    },
    "platform": "azure",
    "azureDevOps": {
        "apiBaseUrl": "https://dev.azure.com/ORGANIZATION/",
        "gitBaseUrl": "https://dev.azure.com/ORGANIZATION/PROJECT/_git/REPO"
    },
    "repositories": [
        "REPO"
    ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```

"loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "USQh_rvb928xD5oGMYaEW",
           "repository": "PROJECT/REPO",
           "err": {
             "task": {
               "commands": [
                 "push",
                 "origin",
                 "renovate/configure:renovate/configure",
                 "--force-with-lease",
                 "-u",
                 "--no-verify",
                 "--verbose",
                 "--porcelain"
               ],
               "format": "utf-8",
               "parser": "[function]"
             },
             "message": "To https://**redacted**@whitesourcesoftware.com' which does not match the policy-specified patterns. )\nDone\nPushing to https://**redacted**@whitesourcesoftware.com' which does not match the policy-specified patterns. \nerror: failed to push some refs to 'https://dev.azure.com/ORGANIZATION/PROJECT/_git/REPO'\n",
             "stack": "Error: To https://**redacted**@whitesourcesoftware.com' which does not match the policy-specified patterns. )\nDone\nPushing to https://**redacted**@whitesourcesoftware.com' which does not match the policy-specified patterns. \nerror: failed to push some refs to 'https://dev.azure.com/ORGANIZATION/PROJECT/_git/REPO'\n\n    at Object.action (/Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/runner/.npm/_npx/59c0475bfd22776c/node_modules/simple-git/dist/cjs/index.js:55:24)"
           },
           "msg": "Repository has unknown error"
         }
       ]

```

</details>

**Comment by rarkins** [comment, marked as answer]
I'm not sure if the RenovateMe extension is still maintained or working, and I'm not familiar with its config

**Reply by MarkFl12** [reply]
I'm having this exact same issue running against Azure Devops from Docker?

**Comment by crookm** [comment]
I recently came across this as well, with a similar configuration except for use of the `Renovate ME` extension in Azure DevOps.

This is not an issue with Renovate (or the DevOps extension), I found it was due to the repository policy configured in Azure DevOps to block commits by git author emails which do not match the configured patterns. The policy is called [commit author email validation](https://learn.microsoft.com/en-us/azure/devops/repos/git/repository-settings?view=azure-devops&tabs=browser#commit-author-email).

From what I could see in the [source](https://github.com/renovatebot/renovate/blob/8b481a11d53d44c04c286699dc64136d63e336e2/lib/util/git/index.ts#L283), it looks like the default git author is `Renovate Bot <renovate@whitesourcesoftware.com>`.

**The solution** is to either permit the default git author in the DevOps policy, or to configure the [gitAuthor ](https://docs.renovatebot.com/configuration-options/#gitauthor) to something that would pass the policy.
